File: frontend/types/project.ts
export interface ProjectDto {
  id: string
  name: string
  description: string | null
  gitHubRepoUrl: string
  maskedGitHubPat: string
  createdAt: string
  updatedAt: string
}

export interface CreateProjectRequest {
  name: string
  description?: string
  gitHubRepoUrl: string
  gitHubPat: string
}

export interface UpdateProjectRequest {
  name: string
  description?: string
  gitHubRepoUrl: string
  gitHubPat?: string | null
}

File: backend/src/DrimAgents.Api/Domain/Projects/Project.cs
using DrimAgents.Api.Domain.Users;

namespace DrimAgents.Api.Domain.Projects;

public class Project
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public required string GitHubRepoUrl { get; set; }
    public required string EncryptedGitHubPat { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public User User { get; set; } = null!;
}
